File: Projeto_8/models/disciplina.py
class Disciplina:
    
    """Cria o construtor da disciplina"""
    def __init__(self, nome, carga_horaria):
        self.nome = nome
        self.carga_horaria = carga_horaria
        self.__nota = 0.0  # Atributo privado

    """Valida a nota"""
    def set_nota(self, nota):
        if 0.0 <= nota <= 10.0:
            self.__nota = nota
        else:
            print("Nota inválida. Deve estar entre 0 e 10.")

    def get_nota(self):
        return self.__nota

File: disciplina.cpp
#include "Disciplina.h"

Disciplina::Disciplina() {
    nome = "";
    cargaHoraria = 0;
    nota = 0;
}

Disciplina::Disciplina(string nome, int cargaHoraria, float nota) {
    this->nome = nome;
    this->cargaHoraria = cargaHoraria;
    this->nota = nota;
}

void Disciplina::setNota(float nota) {
    this->nota = nota;
}

float Disciplina::getNota() {
    return nota;
}

bool verificarAprovacao(Disciplina d) {
    return d.nota >= 6.0;
}
